packages/core/src/domain/conversion/bundle-modifications/__tests__/post-process-bundle.test.ts

```diff
@@ -0,0 +1,225 @@
+      expect(updatedBundle).toEqual(bundle);
+    });
+  });
+
+  describe("addMissingRequests", () => {
+    it("appends a PUT request to each BundleEntry with correct resource references", () => {
+      const { patientId, condition } = initTest();
+      const condition2 = makeCondition({ id: faker.string.uuid() }, patientId);
+      const bundle = makeBundle({ entries: [condition, condition2], type: "collection" });
+
+      const updatedBundle = bundleMods.addMissingRequests(bundle);
+      expect(updatedBundle.entry).toHaveLength(2);
+
+      updatedBundle.entry?.forEach((entry, index) => {
+        const resource = bundle.entry?.[index]?.resource;
+        expect(entry.request).toEqual({
+          method: "PUT",
+          url: `${resource?.resourceType}/${resource?.id}`,
+        });
+      });
+    });
+
+    it("returns bundle unchanged when no entries exist", () => {
+      const bundle = makeBundle({ entries: [], type: "collection" });
+      const updatedBundle = bundleMods.addMissingRequests(bundle);
+      expect(updatedBundle).toEqual(bundle);
+    });
+  });
+
+  describe("removePatientFromConversion", () => {
+    it("throws error when multiple Patient resources exist", () => {
+      const { patientId, condition } = initTest();
+      const patient1: Patient = {
+        resourceType: "Patient",
+        id: patientId,
+        name: [{ given: ["John"], family: "Doe" }],
+      };
+      const patient2: Patient = {
+        resourceType: "Patient",
+        id: faker.string.uuid(),
+        name: [{ given: ["Jane"], family: "Doe" }],
+      };
+      const bundle = makeBundle({ entries: [condition, patient1, patient2], type: "collection" });
+
+      expect(() => bundleMods.removePatientFromConversion(bundle)).toThrow(
+        "Multiple Patient resources found in Bundle"
+      );
+      expect(bundle.entry).toHaveLength(3);
+    });
+
+    it("removes single Patient resource from the Bundle", () => {
+      const { patientId, condition } = initTest();
+      const patient: Patient = {
+        resourceType: "Patient",
+        id: patientId,
+        name: [{ given: ["John"], family: "Doe" }],
+      };
+      const bundle = makeBundle({ entries: [condition, patient], type: "collection" });
+
+      const updatedBundle = bundleMods.removePatientFromConversion(bundle);
+      expect(updatedBundle.entry).toHaveLength(1);
+      expect(updatedBundle.entry?.[0]?.resource?.resourceType).toBe("Condition");
+      expect(bundle.entry).toHaveLength(2);
+    });
+
+    it("returns bundle unchanged when no entries exist", () => {
+      const bundle = makeBundle({ entries: [], type: "collection" });
+      const updatedBundle = bundleMods.removePatientFromConversion(bundle);
+      expect(updatedBundle).toEqual(bundle);
+    });
+  });
+
+  describe("postProcessBundle", () => {
+    it("calls all required processing functions in correct order", () => {
```

Comment:
To check they were called in the right order we would need to check the params sent to each one of them, right?

This can be done checking `.mock.calls[..][..]` - search for `[0][0]` in the code base to see examples.

Reply:
I checked it out, and decided to take another approach for the order of execution check, which seems more straightforward